src/panels/CreateClusterPanel.ts

```diff
@@ -323,6 +335,44 @@ async function createCluster(
     }
 }
 
+async function doFeatureRegistration(preset: string, featureClient: FeatureClient) {
+    if (preset !== "automatic") {
+        return;
+    }
+    const features: MultipleFeatureRegistration[] = [
+        {
+            resourceProviderNamespace: "Microsoft.ContainerService",
+            featureName: "EnableAPIServerVnetIntegrationPreview",
+        },
+        {
+            resourceProviderNamespace: "Microsoft.ContainerService",
+            featureName: "NRGLockdownPreview",
```

Comment:
💡 I am guessing all this is needed for this feature to be enabled, might be add the doc link at the top as comment in the code please.